server/src/diagnostics.ts

```diff
@@ -237,6 +269,24 @@ function maybeAddCriticDiag(violation: string, diagnostics: Diagnostic[], settin
     });
 }
 
+function maybeAddPerlImportsDiag(violation: string, diagnostics: Diagnostic[], settings: NavigatorSettings): void {
+    const diag = JSON.parse(violation);
```

Comment:
`JSON.parse` will throw an error on invalid JSON so I always wrap it in a try/catch.

There's an assumption that the parsed json will have all expected properties. You could wrap the whole thing in the try/catch, that way if any if the expected properties don't exist we will avoid an error.

If you wanted to get really fancy, you could create a function or class that parses the JSON and validates that all the expected properties exist. Personally, I would just use a try/catch and swallow the error and do nothing if something goes wrong since there is nothing we can do if the JSON is invalid anyway.